cmd: Use genericclioptions

mauriciovasquezbernal opened the pull request:
genericclioptions is helper package that makes it easier to develop plugins
that offer a similar experience to kubectl.

This commit uses this package, the main change is that the custom handling of
the namespace is removed everywhere. As a result of this, commands that doesn't
specify a namespace act on the configured namespace in the kubeconfig file.

Part of https://github.com/kinvolk/inspektor-gadget/issues/98.

alban commented:
Does it also work for in-cluster kubeconfig?

mauriciovasquezbernal commented:
> Does it also work for in-cluster kubeconfig?

Yes, it does. I tried it by running `kubectl-gadget` inside the gadget container.

alban commented:
Could we have a `-A` / `--all-namespaces` option? (there is one in traceloop but not the other bcc-based gadgets, as far as I understand)

For the `profile` gadget, it would be useful to be able to trace on the host. I think this patch removes that possibility? Could we have a `--host` option? I guess that only makes sense for the `profile` gadget.

mauriciovasquezbernal commented:
I pushed a change to support `-A, --all-namespaces` option. Before this PR it was not possible to track events on the host, it can be easily implemented. I'll open an issue for that.

I also did some small changes to the flags, added short options `-l`, `-c`, and `-p` and renamed `--label` to `--selector` to make it similar to `kubectl`. Please let me know what do you think about it.

alban commented on cmd/kubectl-gadget/bcck8s.go:
```diff
@@ -97,18 +95,51 @@ func init() {
+			"selector",
+			"l",
+			"",
+			fmt.Sprintf("Labels selector to filter on. Only '=' is supported.(e.g. key1=value1)."),
```
`s/.(/ (/`

alban replied:
You could add two labels in the example to show they are separated by `,`:
`-l key1=value1,key2=value2`

mauriciovasquezbernal replied:
Done!

alban commented on cmd/kubectl-gadget/main.go:
```diff
@@ -2,25 +2,26 @@ package main
-		"Path to kubeconfig file")
-	viper.BindPFlag("kubeconfig", rootCmd.PersistentFlags().Lookup("kubeconfig"))
+	KubernetesConfigFlags.AddFlags(rootCmd.PersistentFlags())
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```
Do we have any flag with a `-` or environment variable with a `_`?

mauriciovasquezbernal replied:
There are some flags from the `genericclioptions` package with `-` like `as-group`, `certificate-authority`, `client-certificate`, etc.
It was suggested to be used in the plugin example.
